### ISSUE TITLE
ignore_class_list was a breaking addition

### ISSUE THREAD

**Issue by Pomax** [Pomax/react-onclickoutside#209]
@Andarist adding the ignore_class_list to the exports means that `var hoc = require('react-onclickoutside')` no longer works, as that require call now yields an object
```js
{ __esModule: true,
  IGNORE_CLASS_NAME: 'ignore-react-onclickoutside',
  default: [Function: onClickOutsideHOC]
}
```
so all the code in the readme that explains how to use this library is now wrong. I am backing out the ignore class commit, and if we want to reintroduce, that will be a breaking change warranting a v7

**Comment by Pomax** [comment]
I thought we'd fixed the repo so that it would run an ES conversion for people using node "the old way", but I see node requiring in the ES6+ version when doing a normal `require` vs using `import` statements. 

I'd like to know what is going on here.

**Comment by Andarist** [comment]
**TL;DR**
v6 broke this and not that merge. When using bare `require` it indeed now needs to access `default` property like this `require('react-onclickoutside').default`. If docs use bare `require` in its examples they need to be:
- refactored to es6 modules
- refactored to `.default`

**Full explanation**
Aint sure how did u check this - plain node.js? If so, please also check previous version as generally it should yield **the same** result.

When importing through `babel` it wraps required modules like this:
```js
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var _reactOnclickoutside = require('react-onclickoutside');
var _reactOnclickoutside2 = _interopRequireDefault(_reactOnclickoutside);
```

So it is able to unwrap properly what got exported after transpilation.

Please take a look at what is exported now (and before `IGNORE_CLASS_NAME` merge).
```js
exports.__esModule = true;
exports.default = onClickOutsideHOC;

```
And what is exported with `IGNORE_CLASS_NAME` merged in:
```js
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.IGNORE_CLASS_NAME = undefined;
exports.default = onClickOutsideHOC;
var IGNORE_CLASS_NAME = exports.IGNORE_CLASS_NAME = 'ignore-react-onclickoutside';
```

So what u should notice, structure is basically the same and shouldnt matter.

**Comment by Pomax** [comment]
Secondary note: https://github.com/Pomax/react-onclickoutside/issues/210 is not *directly* related, but seems to have been caused by the deps cleanup

**Comment by Andarist** [comment]
Closing this, gonna rerelease discussed code in following days.
